pkgs/development/libraries/gts/default.nix

```diff
@@ -1,34 +1,65 @@
+{ fetchurl, stdenv, pkgconfig, autoreconfHook, gettext, glib, buildPackages }:
 
+# gts generates binaries during build that will be used during the build process.
+# In case of cross-platform  builds, this package needs to bootstrap itself
+# in order to generate these binaries for the propper build architecture.
+# Please keep that in mind when making changes to the derivation.
 
-stdenv.mkDerivation rec {
-  pname = "gts";
-  version = "0.7.6";
+# When making significant changes, please try to test cross-platform builds as well.
+# Example:
+#   nix-build . -A pkgsCross.aarch64-multiplatform.gts
+#   nix-shell -p qemu --run "qemu-aarch64 result-bin/bin/transform --help"
 
-  outputs = [ "bin" "dev" "out" ];
+let
 
-  src = fetchurl {
-    url = "mirror://sourceforge/gts/${pname}-${version}.tar.gz";
-    sha256 = "07mqx09jxh8cv9753y2d2jsv7wp8vjmrd7zcfpbrddz3wc9kx705";
-  };
+  cross = stdenv.hostPlatform != stdenv.buildPlatform;
 
-  nativeBuildInputs = [ pkgconfig autoreconfHook ];
-  buildInputs = [ gettext ];
-  propagatedBuildInputs = [ glib ];
+  name = "gts";
+  drv = { pname ? name } : rec {
+    inherit pname;
+    version = "0.7.6";
 
-  doCheck = false; # fails with "permission denied"
+    outputs = [ "bin" "dev" "out" ];
 
-  meta = {
-    homepage = "http://gts.sourceforge.net/";
-    license = stdenv.lib.licenses.lgpl2Plus;
-    description = "GNU Triangulated Surface Library";
+    depsBuildBuild = [ buildPackages.stdenv.cc ];
 
-    longDescription = ''
-      Library intended to provide a set of useful functions to deal with
-      3D surfaces meshed with interconnected triangles.
-    '';
+    src = fetchurl {
+      url = "mirror://sourceforge/gts/${name}-${version}.tar.gz";
+      sha256 = "07mqx09jxh8cv9753y2d2jsv7wp8vjmrd7zcfpbrddz3wc9kx705";
+    };
+
+    # glib needs to be in nativeBuildInputs in order for AM_PATH_GLIB_2_0
+    # c to be generated orrectly during cross-platform builds
+    nativeBuildInputs = [ pkgconfig autoreconfHook glib ];
+    buildInputs = [ gettext ];
+    propagatedBuildInputs = [ glib ];
+
+    doCheck = false; # fails with "permission denied"
+
+    meta = {
+      homepage = "http://gts.sourceforge.net/";
+      license = stdenv.lib.licenses.lgpl2Plus;
+      description = "GNU Triangulated Surface Library";
 
-    maintainers = [ stdenv.lib.maintainers.viric ];
-    platforms = stdenv.lib.platforms.linux ++ stdenv.lib.platforms.darwin;
+      longDescription = ''
+        Library intended to provide a set of useful functions to deal with
+        3D surfaces meshed with interconnected triangles.
+      '';
+
+      maintainers = [ stdenv.lib.maintainers.viric ];
+      platforms = stdenv.lib.platforms.linux ++ stdenv.lib.platforms.darwin;
+    };
   };
-}
+
+  bootstrap = stdenv.mkDerivation (drv { pname = "${name}-bootstrap"; } // stdenv.lib.optionalAttrs (cross) {
+    configureFlags = [ "CC=${buildPackages.stdenv.cc}/bin/cc" ];
```

Comment:
Would not it perhaps be possible to just change the compiler for building the `predicates_init` in the `Makefile.am`? I am not very familiar with automake myself but this looks overly complicated.